operator/pkg/securedcluster/scanner/auto_sense.go

```diff
@@ -21,14 +30,39 @@ func AutoSenseLocalScannerSupport(ctx context.Context, client ctrlClient.Client,
 		if err != nil {
 			return false, errors.Wrap(err, "detecting presence of a Central CR in the same namespace")
 		}
-		return !siblingCentralPresent, nil
+		isOpenShift, err := isRunningOnOpenShift(ctx, client, s.GetNamespace())
+		if err != nil {
+			return false, errors.Wrap(err, "cannot fetch OpenShift ClusterVersion resource")
+		}
+		enableScanner := isOpenShift && !siblingCentralPresent
+		return enableScanner, nil
 	case platform.LocalScannerComponentDisabled:
 		return false, nil
 	}
 
 	return false, errors.Errorf("invalid spec.scanner.scannerComponent %q", scannerComponent)
 }
 
+func isRunningOnOpenShift(ctx context.Context, client ctrlClient.Client, namespace string) (bool, error) {
```

Comment:
I think the `namespace` param is not necessary. The `ClusterVersion` resource is not namespaced.
See: https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L8

FYI: There are resources cluster scoped like `ClusterRole` or `Namespace` which do not fit in Kubernetes namespace model. These can't be separated by namespace, projects like [kcp](https://github.com/kcp-dev/kcp) try to improve this situation.
For us it is mostly interesting because of resource name clashes (i.e. having two global resource with the same name clash) and if we can omit the namespace in client calls.

If you want to double check if the resource is globally scoped you can also run the `kubectl get ` command against any namespace and should always return the same result.

Reply:
You are right `describe` says that it has no namespace:
```
kubectl describe clusterversions.config.openshift.io
Name:         version
Namespace:
Labels:       <none>
...
```